pkg/apis/pipeline/v1/taskref_types.go

```diff
@@ -48,5 +50,5 @@ const (
 func (tr *TaskRef) IsCustomTask() bool {
 	// Note that if `apiVersion` is set to `"tekton.dev/v1beta1"` and `kind` is set to `"Task"`,
 	// the reference will be considered a Custom Task - https://github.com/tektoncd/pipeline/issues/6457
-	return tr != nil && tr.APIVersion != "" && tr.Kind != ""
+	return tr != nil && tr.APIVersion != "" && (tr.Kind != "" || tr.CustomTask != "")
```

Comment:
Is this modification necessary? Since we are setting Kind and APiVersion already, can we just continue to rely on existing behavior?